deepracer/logs/log.py

```diff
@@ -0,0 +1,161 @@
+import glob
+import json
+import os
+import pandas as pd
+import re
+from joblib import Parallel, delayed
+
+
+class DeepRacerLog:
+    def __init__(self):
+        # Column names we support in the CSV file.
+        self.col_names = [
+            "episode",
+            "steps",
+            "x",
+            "y",
+            "heading",
+            "steering_angle",
+            "speed",
+            "action",
+            "reward",
+            "done",
+            "all_wheels_on_track",
+            "progress",
+            "closest_waypoint",
+            "track_len",
+            "tstamp",
+            "episode_status",
+        ]
+        self.hyperparam_keys = [
+            "batch_size",
+            "beta_entropy",
+            "e_greedy_value",
+            "epsilon_steps",
+            "exploration_type",
+            "loss_type",
+            "lr",
+            "num_episodes_between_training",
+            "num_epochs",
+            "stack_size",
+            "term_cond_avg_score",
+            "term_cond_max_episodes"
+        ]
+
+    def load(self):
+        """Method that loads a DeepRacer log into a dataframe.
+
+        Raises:
+            NotImplementedError This class should not be used.
+        """
+        raise NotImplementedError("Implement this in a subclass!")
+
+    def dataframe(self):
+        """Method that provides the dataframe for analysis of this log.
+
+        Raises:
+            NotImplementedError: This class should not be used.
+        """
+        raise NotImplementedError("Implement this in a subclass!")
+
+    def hyperparameters(self):
```

Comment:
`action_space` and `agent_and_network` should also be defined here

Reply:
Done